pkgs/build-support/mkshell/minimal.nix

```diff
@@ -0,0 +1,47 @@
+{ writeTextFile, writeScript, bash }:
+
+# A special kind of derivation that is only meant to be consumed by the
+# nix-shell. This differs from the traditional `mkShell` in that it is
+# ultra minimalist in dependencies.
+# It does not come with traditional stdenv (i.e. coreutils).
+# Its only dependency is essentially bash.
+{
+# a list of packages to add to the shell environment
+# we simplify here rather than `native` or `propagated` since
+# this is only ever used to be in a nix-shell; which is by default
+# the current system
+packages ? [ ], ... }@attrs:
+
+derivation ({
+  name = "minimal-nix-shell";
+
+  # What is the need for this reference to stdenv?
+  # Well Nix itself, checks to see if this attribute exists
+  # and sources the file.
+  # We will setup our nix-shell environment here and do things
+  # like clear the $PATH
+  # reference: https://github.com/NixOS/nix/blob/94ec9e47030c2a7280503d338f0dca7ad92811f5/src/nix-build/nix-build.cc#L494
+  "stdenv" = writeTextFile rec {
+    name = "setup";
+    executable = true;
+    destination = "/${name}";
+    text = ''
+      set -e
+      unset PATH
+
+      for a in $packages; do
+        export PATH=$a/bin:$PATH
+      done
+    '';
+  };
+
+  builder = writeScript "builder.sh" ''
+    #!${bash}/bin/bash
+    echo
+    echo "This derivation is not meant to be built, aborting";
+    echo
+    exit 1
```

Comment:
@Infinisil I actually don't see a particular reason why nix-shells can't be buildable.
I wonder if doing your _export_ trick here to $out is enough + adding an error message.